### PR TITLE
chore: remove stream seek constraint

### PR DESCRIPTION
**Pull request by lemmih**
## Summary of changes

Changes introduced in this pull request:

- Remove `AsyncSeek` constraint for `CarStream`. The seek constraint prevents streaming data from, say, an HTTP server.

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -184,6 +184,7 @@ url = { version = "2.3", features = ["serde"] }
 uuid = { version = "1.3", features = ['v4'] }
 walkdir = "2"
 zstd = "0.12.3"
+zstd-safe = "6.0"
```
Comment (hanabi1224) [inline Cargo.toml:187]: `zstd-safe` has been re-exported from `zstd`

https://docs.rs/zstd-safe/6.0.6/zstd_safe/index.html

**Comment by lemmih** [comment]
It's technically possible to craft an uncompressed CAR file which will be identified as a zstd compressed file. However, no valid _Filecoin_ CAR file can be misidentified in this way.

File: src/utils/db/car_stream.rs [hunk 2]
```diff
@@ -12,10 +12,10 @@ use fvm_ipld_encoding::to_vec;
 use integer_encoding::VarInt;
 use pin_project_lite::pin_project;
 use serde::{Deserialize, Serialize};
-use std::io::{self, Cursor, SeekFrom};
+use std::io;
 use std::pin::Pin;
 use std::task::{Context, Poll};
-use tokio::io::{AsyncBufRead, AsyncRead, AsyncSeek, AsyncSeekExt, AsyncWrite};
```
Comment (hanabi1224) [inline src/utils/db/car_stream.rs:18]: Do we still need `RandomAccessFile` after removing `AsyncSeek` constraint?

Comment (hanabi1224) [reply src/utils/db/car_stream.rs:18]: Nvm it does not seem to be related

**Review by hanabi1224 — APPROVED** [review]

**Review by elmattic — APPROVED** [review]
Do we have a test that exercice the non-compressed path?

**Comment by lemmih** [comment]
> Do we have a test that exercice the non-compressed path?

Added a test using the uncompressed genesis CAR file.

**Pull request merged**